InputValidation/IValidation.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+/// <summary>
+/// Interface for various forms of input validation in TotalImage.
+/// </summary>
+namespace TotalImage.Validation
+{
+    public interface IValidation
+    {
+        object Validate(object Data); // Can return modified data. For example, upper-to-lower case.
+        void OnFail();
+        void OnSuccess();
```

Comment:
I'm not sure these should be in the interface. How to handle a failure or success should be down to the calling code rather than methods on the Validator.